### PR TITLE
Bug/jmv 45 lin regression not working 10000 cases

### PR DESCRIPTION
**Pull request by raviselker**
Fixes jamovi/jamovi-fun-todo-list#45

### CODE REVIEW

File: R/linreg.b.R [hunk 1]
```diff
@@ -110,16 +97,66 @@ linRegClass <- R6::R6Class(
                         anovaTerms=anovaTerms, AIC=AIC, BIC=BIC, CI=CI, CIScaled=CIScaled,
                         dwTest=dwTest, VIF=VIF, cooks=cooks))
         },
+        .computeModels = function(data, model = NULL, scaled = FALSE) {
```
Comment (jonathon-love) [inline R/linreg.b.R]: could this be `modelNo` rather than `model`? `model` makes me expect the model object.

File: R/linreg.b.R [hunk 2]
```diff
@@ -715,27 +742,26 @@ linRegClass <- R6::R6Class(
...
             }
         },
         .qqPlot = function(image, ggtheme, theme, ...) {
 
             if (is.null(image$state))
                 return(FALSE)
 
-            p <- ggplot(data=image$state, aes(x=x, y=y)) +
+            data <- private$.cleanData()
+            model <- private$.computeModels(data, model=image$state)[[1]]
```
Comment (jonathon-love) [inline R/linreg.b.R]: you're better off assigning the models to (something like) `private$.models` in the `.run()` phase, and only computing them here if `private$.models` is unassigned. when the analysis is run, the same analysis object is used to create all the plots. it's only when the user, say, exports an image, that the plot function is called without `.run()` being called first.

you could go further of course, have a caching active binding, etc.

**Review by jonathon-love — CHANGES_REQUESTED** [review]

File: R/linreg.b.R (hunk 2 above)
Comment (raviselker) [reply R/linreg.b.R]: So something like: 
```        
models = function() {

            if (! is.null(private$.models))
                return(private$.models)

            data <- private$.cleanData()
            models <- private$.computeModels(data)

            return(models)
},
```

File: R/linreg.b.R (hunk 1 above)
Comment (raviselker) [reply R/linreg.b.R]: Fixed

File: R/linreg.b.R (hunk 2 above)
Comment (raviselker) [reply R/linreg.b.R]: Fixed this by a major refactoring

**Comment by jonathon-love** [comment]
just a heads up ... i think the next time we push jmv to CRAN, CRAN will insist that all these active bindings are documents :/

**Pull request merged**